src/Mono.Android/Xamarin.Android.Net/OldAndroidSSLSocketFactory.cs

```diff
@@ -0,0 +1,67 @@
+using Java.Net;
+using Javax.Net.Ssl;
+
+namespace Xamarin.Android.Net
+{
+	// Context: https://github.com/xamarin/xamarin-android/issues/1615
+	//
+	// Code based on the code provided in the issue above
+	//
+	class OldAndroidSSLSocketFactory : SSLSocketFactory
+	{
+		readonly SSLSocketFactory factory = (SSLSocketFactory)Default;
+
+		public override string[] GetDefaultCipherSuites ()
+		{
+			return factory.GetDefaultCipherSuites ();
+		}
+
+		public override string[] GetSupportedCipherSuites ()
+		{
+			return factory.GetSupportedCipherSuites ();
+		}
+		public override Socket CreateSocket (InetAddress address, int port, InetAddress localAddress, int localPort)
+		{
+			return EnableTlsOnSocket (factory.CreateSocket (address, port, localAddress, localPort));
+		}
+
+		public override Socket CreateSocket (InetAddress host, int port)
+		{
+			return EnableTlsOnSocket (factory.CreateSocket (host, port));
+		}
+
+		public override Socket CreateSocket (string host, int port, InetAddress localHost, int localPort)
+		{
+			return EnableTlsOnSocket (factory.CreateSocket (host, port, localHost, localPort));
+		}
+
+		public override Socket CreateSocket (string host, int port)
+		{
+			return EnableTlsOnSocket (factory.CreateSocket (host, port));
+		}
+
+		public override Socket CreateSocket (Socket s, string host, int port, bool autoClose)
+		{
+			return EnableTlsOnSocket (factory.CreateSocket (s, host, port, autoClose));
+		}
+
+		public override Socket CreateSocket ()
+		{
+			return EnableTlsOnSocket (factory.CreateSocket ());
+		}
+
+		private Socket EnableTlsOnSocket (Socket socket)
+		{
+			if (socket is SSLSocket sslSocket) {
+				sslSocket.SetEnabledProtocols (sslSocket.GetSupportedProtocols ());
+			}
+			return socket;
+		}
+
+		protected override void Dispose (bool disposing)
+		{
+			factory.Dispose ();
```

Comment:
You should not do this.

The problem is "reference sharing": every `AndroidClientHandler` instance (on API-16..20 devices) will have an `OldAndroidSSLSocketFactory` instance -- so let's call it `N` instances of `AndroidClientHandler` & `OldAndroidSSLSocketFactory` -- *each of which* will refer to the ***same*** `SSLSocketFactory` value returned by `SSLSocketFactory.Default`.

This results in "unintentional spooky action at a distance", a'la:

```
var a = new AndroidClientHandler (...);
var b = new AndroidClientHandler (...);

a.Dispose(); /* ...or otherwise cause `a.httpsConnection.SSLSocketFactory.Dispose()` to be invoked */
/* use `b` */
```

Once `a.httpsConnection.SSLSocketFactory` has been disposed, `b.httpsConnection.SSLSocketFactory` can no longer be used *either*, because even though the `httpsConnection.SSLSocketFactory` instances differ, `OldAndroidSSLSocketFactory.factory` for *all* instances refers to the *same* instance, and when that instance was disposed from `a`, `b` can no longer use it.